features/machine/autoscaler.feature

```diff
@@ -52,3 +52,134 @@ Feature: Cluster Autoscaler Tests
+    And evaluation of `machine_set.name` is stored in the :machineset_clone0_name clipboard
+    Given I clone a machineset named "machineset-clone1"
+    And evaluation of `machine_set.name` is stored in the :machineset_clone1_name clipboard
+    Given I clone a machineset named "machineset-clone2"
+    And evaluation of `machine_set.name` is stored in the :machineset_clone2_name clipboard
+
+    When I run oc create over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/cloud/machine-autoscaler.yml" replacing paths:
+      | ["metadata"]["name"]                   | maotest0                                                    |
+      | ["spec"]["scaleTargetRef"]["name"]     | <%= cb.machineset_clone0_name %>                            |
+    Then the step should succeed
+    When I run oc create over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/cloud/machine-autoscaler.yml" replacing paths:
+      | ["metadata"]["name"]                   | maotest1                                                    |
+      | ["spec"]["scaleTargetRef"]["name"]     | <%= cb.machineset_clone1_name %>                            |
+    Then the step should succeed
+    
+    When I run the :patch admin command with:
+      | resource          | machineautoscaler                                                                |
+      | resource_name     | maotest0                                                                         |
+      | p                 | {"spec":{"scaleTargetRef":{"name":"<%= cb.machineset_clone2_name %>"}}}          |
+      | type              | merge                                                                            |
+    Then the step should succeed
+    When I run the :describe admin command with:
+      | resource          | machineautoscaler                                                                |
+      | name              | maotest0                                                                         |
+    Then the step should succeed
+    And the output should match "Name:\s+<%= cb.machineset_clone2_name %>"  
+    When I run the :describe admin command with:
+      | resource          | machineset                                                                       |
+      | name              | <%= cb.machineset_clone0_name %>                                                 |
+    Then the step should succeed
+    And the output should match "Annotations:\s+<none>"
+    When I run the :describe admin command with:
+      | resource          | machineset                                                                       |
+      | name              | <%= cb.machineset_clone2_name %>                                                 |
+    Then the step should succeed
+    And the output should match "Annotations:\s+autoscaling.openshift.io/machineautoscaler: openshift-machine-api/maotest0"
+  
+    When I run the :patch admin command with:
```

Comment:
Do the following tests duplicate the previous? I think you only need 2 machinesets and 1 machineautoscaler to cover the scenario.

Reply:
This case tests 2 scenarios:  
1) update a machineautoscaler to reference a  MachineSet not owned by another MachineAutoscaler
2) update a machineautoscaler to reference a  MachineSet already owned by another MachineAutoscaler
So 2 machineautuscalers are necessary, if we clone 2 machinesets we can cover the 2 scenarios by:
create mao1->machineset1
update mao1->machineset2
check result
create mao2->machineset1
update mao->machineset1
check result

Reply:
I get it, thank you.